k8s/upgrade/src/plugin/upgrade.rs

```diff
@@ -833,3 +931,62 @@ pub(crate) async fn get_source_version(ns: &str) -> error::Result<String> {
         .to_string();
     Ok(value.to_string())
 }
+
+/// Parse set-file and create config map data.
+pub(crate) async fn create_config_map_data(
+    upgrade_args: &UpgradeArgs,
+) -> error::Result<BTreeMap<String, String>> {
+    let mut upgrade_map = BTreeMap::new();
+    for file in &upgrade_args.set_file {
+        let data: Vec<_> = file.split('=').collect();
+        let file_name = Path::new(data[1])
+            .file_name()
+            .unwrap_or_default()
+            .to_owned()
+            .into_string()
+            .unwrap();
+        let cm_values =
+            fs::read_to_string(data[1]).context(error::ReadFromFile { filepath: data[1] })?;
+        let set_param = data[0].to_string();
+        // create config map key with key of set-file plus the file name
+        // this is done to make the key unique as /root/A.yaml and /root/data/A.yaml are different.
+        // Example : -- set-file jaeger-operator.tolerations=/root/tolerations.yaml
+        // create the key as jaeger-operator.tolerations_tolerations.yaml
+        // this is acts as a name of file under the mounted path inside pod.
+        let cm_key = format!("{set_param}_{file_name}");
```

Comment:
There should no more configmap data entries than the number of files. The same file should not create multiple configmap data entries.

Reply:
there can be scenarios:
1. --set-flag  jaeger-operator.tolerations=/root/tolerations.yaml --set-file ha.node.tolerations=/root/new/tolerations.yaml
Here the name of both the files are same but the files are different . So the `jaeger-operator.tolerations_tolerations.yaml`  and `ha.node.tolerations_tolerations.yaml` are two different files created in the pods.
2.  --set-flag  jaeger-operator.tolerations=/root/tolerations.yaml --set-file ha.node.tolerations=/root/tolerations.yaml. the files are passed are same. The PR still will create two file in pod. 
The problem here is we cant use the complete path `/root/tolerations.yaml` as the key as `/` is not allowed as the key for configmap. The key becomes the name of the file inside pod. Here the allowed values are `.`, `-`, `_`. So we had to parse `/root/tolerations.yaml` as `-root-tolerations.yaml` and then use it . This parsing is not ideal as the folders can have any seperators.

Solution to this problem : 
 a - just take the file name like `tolerations.yaml` as key . In this case both `/root/tolerations.yaml ` and `/root/new/tolerations.yaml` will have `tolerations.yaml`  as key , but the since data is a BtreeMap , the former gets overwritten by later , so even there are two different file , only one file gets created inside pod.
b- add a random number and append it to the file name to make it unique. In this case too two files will be generated. This has been done in the PR ,not by adding the random number, but appending the key `jaeger-operator.tolerations` to file name and thus making it unique.

IMHO this is the ideal approach where we dont have to do any parsing on the upgrade job . Its just running helm upgrade with `--set-flag  jaeger-operator.tolerations=/root/tolerations.yaml --set-file ha.node.tolerations=/root/new/tolerations.yaml` flag.

Reply:
Why can't we just use the key as the cm_key?

Reply:
we can use it , updated the code to use just key.